code/modules/mob/living/carbon/carbon_update_icons.dm

```diff
@@ -216,84 +218,104 @@
 
 	. += emissive_blocker(standing.icon, standing.icon_state, alpha = standing.alpha)
 
-/mob/living/carbon/update_body()
-	update_body_parts()
-
-/mob/living/carbon/proc/assign_bodypart_ownership()
-	for(var/X in bodyparts)
-		var/obj/item/bodypart/BP = X
-		BP.original_owner = WEAKREF(src)
+/mob/living/carbon/update_body(is_creating)
+	update_body_parts(is_creating)
 
-/mob/living/carbon/proc/update_body_parts()
-	//CHECK FOR UPDATE
-	var/oldkey = icon_render_key
-	icon_render_key = generate_icon_render_key()
-	if(oldkey == icon_render_key)
+///Checks to see if any bodyparts need to be redrawn, then does so.
+/mob/living/carbon/proc/update_body_parts(update_limb_data)
+	update_damage_overlays()
+	var/list/needs_update = list()
+	var/limb_count_update = FALSE
+	for(var/obj/item/bodypart/limb as anything in bodyparts)
+		limb.update_limb(is_creating = update_limb_data) //Update limb actually doesn't do much, get_limb_icon is the cpu eater.
+		var/old_key = icon_render_keys?[limb.body_zone] //Checks the mob's icon render key list for the bodypart
+		icon_render_keys[limb.body_zone] = (limb.is_husked) ? limb.generate_husk_key().Join() : limb.generate_icon_key().Join() //Generates a key for the current bodypart
```

Comment:
I don't understand the `Join`s here. Why not just check if list is empty?

Also you can do `condiitonal1 == conditional2` instead of this duped code.

Reply:
`.Join()` converts them to a string, which is the `exact key` we are checking for. If the key is different in *any way*, it means we need to redraw the limb.